extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java

```diff
@@ -245,6 +254,20 @@ private static void contributeClassMarkedForCascadingValidation(Set<DotName> cla
         }
     }
 
+    private static void contributeMethodsWithInheritedValidation(Map<DotName, Set<String>> inheritedAnnotationsToBeValidated,
+            IndexView indexView, MethodInfo method) {
+        ClassInfo clazz = method.declaringClass();
+        if (Modifier.isInterface(clazz.flags())) {
+            // Remember annotated interface methods that must be validated
+            Set<String> methodSet = inheritedAnnotationsToBeValidated.get(clazz.name());
+            if (methodSet == null) {
+                methodSet = new HashSet<String>();
+                inheritedAnnotationsToBeValidated.put(clazz.name(), methodSet);
+            }
```

Comment:
could collapse this down a bit using `computeIfAbsent()`

Reply:
it isn't a concurrent set